common/component/kafka/kafka.go

```diff
@@ -166,6 +223,128 @@ func (k *Kafka) Close() (err error) {
 	return err
 }
 
+func getSchemaSubject(topic string) string {
+	// For now assumes that subject is named after topic (e.g. `my-topic-value`)
+	return fmt.Sprintf("%s-value", topic)
+}
+
+func (k *Kafka) DeserializeValue(message *sarama.ConsumerMessage, config SubscriptionHandlerConfig) ([]byte, error) {
+	if config.ValueSchemaType == Avro {
+		srClient, err := k.getSchemaRegistyClient()
+		if err != nil {
+			return nil, err
+		}
+
+		if len(message.Value) < 5 {
+			return nil, fmt.Errorf("value is too short")
+		}
+		schemaID := binary.BigEndian.Uint32(message.Value[1:5])
+		schema, err := srClient.GetSchema(int(schemaID))
+		if err != nil {
+			return nil, err
+		}
+		// The data coming through is standard JSON. The version currently supported by srclient doesn't support this yet
+		// Use this specific codec instead.
+		codec, err := goavro.NewCodecForStandardJSONFull(schema.Schema())
+		if err != nil {
+			return nil, err
+		}
+		native, _, err := codec.NativeFromBinary(message.Value[5:])
+		if err != nil {
+			return nil, err
+		}
+		value, err := codec.TextualFromNative(nil, native)
+		if err != nil {
+			return nil, err
+		}
+
+		return value, nil
+	}
+	return message.Value, nil
+}
+
+func (k *Kafka) getLatestSchema(topic string) (*srclient.Schema, *goavro.Codec, error) {
+	srClient, err := k.getSchemaRegistyClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	subject := getSchemaSubject(topic)
+	if k.schemaCachingEnabled {
+		cacheEntry, ok := k.latestSchemaCache[subject]
+		// Cache present and not expired
+		if ok && cacheEntry.expirationTime.Compare(time.Now()) < 0 {
```

Comment:
Nit: please use [`time.Before`](https://pkg.go.dev/time#Time.Before) (or `time.After`) for readability